scene/main/instance_placeholder.cpp

```diff
@@ -161,19 +161,20 @@ void InstancePlaceholder::set_value_on_instance(InstancePlaceholder *p_placehold
 	}
 
 	switch (current_type) {
-		case Variant::Type::NIL:
-			if (placeholder_type != Variant::Type::NODE_PATH) {
+		case Variant::Type::NIL: {
+			Ref<Resource> resource = p_set.value;
+			if (placeholder_type != Variant::Type::NODE_PATH && !resource.is_valid()) {
 				break;
 			}
-			// If it's nil but we have a NodePath, we guess what works.
-
+			// If it's nil but we have a NodePath or a Resource, we guess what works.
 			p_instance->set(p_set.name, p_set.value, &is_valid);
 			if (is_valid) {
 				break;
 			}
 
 			p_instance->set(p_set.name, try_get_node(p_placeholder, p_instance, p_set.value), &is_valid);
 			break;
+		}
 		case Variant::Type::OBJECT:
```

Comment:
Could you add braces for this one too while at it, for consistency?

Reply:
No problem. I also should encloses the default or that is unnecessary?

Reply:
Yeah let's do it too for consistency.

Reply:
Done